### PR TITLE
DBZ-6669 Snapshot will not capture data when signal.data.collection is present without table.include.list

### PR DESCRIPTION
**Pull request by mfvitale**
closes: https://issues.redhat.com/browse/DBZ-6669

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java [hunk 1]
```diff
@@ -241,13 +242,16 @@ private Stream<TableId> toTableIds(Set<TableId> tableIds, Pattern pattern) {
...
         String tableIncludeList = connectorConfig.tableIncludeList();
         String signalingDataCollection = connectorConfig.getSignalingDataCollectionId();
         List<Pattern> captureTablePatterns = new ArrayList<>();
         if (!Strings.isNullOrBlank(tableIncludeList)) {
             captureTablePatterns.addAll(Strings.listOfRegex(tableIncludeList, Pattern.CASE_INSENSITIVE));
         }
+        else {
+            captureTablePatterns.add(MATCH_ALL_PATTERN);
+        }
         if (!Strings.isNullOrBlank(signalingDataCollection)) {
```
Comment (Naros) [inline debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java:255]: So this larger block:
```
        if (!Strings.isNullOrBlank(tableIncludeList)) {
            captureTablePatterns.addAll(Strings.listOfRegex(tableIncludeList, Pattern.CASE_INSENSITIVE));
        }
        else {
            captureTablePatterns.add(MATCH_ALL_PATTERN);
        }
        if (!Strings.isNullOrBlank(signalingDataCollection)) {
            captureTablePatterns.addAll(getSignalDataCollectionPattern(signalingDataCollection));
        }
        if (captureTablePatterns.size() > 0) {
            return captureTablePatterns
                    .stream()
                    .flatMap(pattern -> toTableIds(capturedTables, pattern))
                    .collect(Collectors.toCollection(LinkedHashSet::new));
        }
```
I believe this larger code will always go into the `captureTablePatterns.size() > 0` branch now, right?

**Review by Naros — COMMENTED** [review]
Just one comment inline @mfvitale.

File: debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java (hunk 1 above)
Comment (mfvitale) [reply debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java:258]: Yes, good point.

**Review by Naros — APPROVED** [review]

**Comment by Naros** [comment]
Applied to main (=2.4), thanks @mfvitale.

**Pull request merged**